### ISSUE TITLE
[Kosmos-G] Issue for installing with included `torchscale`

### ISSUE THREAD

**Issue by apolinario** [microsoft/unilm#1335]
**Describe the bug**
Trying to install Kosmos-G, there's a disclaimer that says: 

> Make sure you have Pytorch 1.13.0 installed.

However on 

```shell 
bash vl_setup.sh
```
We got 
```shell
pip install torchscale/
```
And in the included torchscale `setup.py` there are two issues with `install_requires`:  https://github.com/microsoft/unilm/blob/master/kosmos-g/torchscale/setup.py#L20 

1. `install_requires` torch>=1.8 seems to contradict the `1.13.0` version requested
2. `install_requires` apex installs the wrong apex package, as the pypi apex has **nothing to do** with Nvidia apex

The problem arises when using:
* [x] the official example scripts
* [ ] my own modified scripts

**Comment by fikry102** [comment]
a related error after running "bash runapp.sh"

from pyramid.session import UnencryptedCookieSessionFactoryConfig
ImportError: cannot import name 'UnencryptedCookieSessionFactoryConfig' from 'pyramid.session' (unknown location)

**Comment by apolinario** [comment]
Yes, that error is directly related as it's trying to execute `apex` from pypi that has nothing to do with Nvidia APEX. One has to uninstall `apex` and then install NVidia Apex https://github.com/NVIDIA/apex

**Comment by fikry102** [comment]
Could you run bash runapp.sh successfully?
Is it right to use the following command?
```
git clone https://github.com/NVIDIA/apex
cd apex
# if pip >= 23.1 (ref: https://pip.pypa.io/en/stable/news/#v23-1) which supports multiple `--config-settings` with the same key... 
pip install -v --disable-pip-version-check --no-cache-dir --no-build-isolation --config-settings "--build-option=--cpp_ext" --config-settings "--build-option=--cuda_ext" ./
# otherwise
pip install -v --disable-pip-version-check --no-cache-dir --no-build-isolation --global-option="--cpp_ext" --global-option="--cuda_ext" ./
```

**Comment by xichenpan** [comment]
Hi @apolinario @fikry102 , thanks for your attention. Sorry fot that, we only test the environment for the docker option, can you try to setup the environment within the docker? Feel free to at me if there is any further questions :D

**Comment by caicj15** [comment]
If anyone can offer docker image for kosmos-g, it will help me a lot.

**Comment by fikry102** [comment]
> Hi @apolinario @fikry102 , thanks for your attention. Sorry fot that, we only test the environment for the docker option, can you try to setup the environment within the docker? Feel free to at me if there is any further questions :D

`sudo docker run --runtime=nvidia --gpus all --ipc=host --ulimit memlock=-1 --ulimit stack=67108864 --name kosmosg --privileged=true -it -v /mnt:/mnt/ nvcr.io/nvidia/pytorch:22.10-py3 /bin/bash`

It gives：
```
docker: Error response from daemon: Unknown runtime specified nvidia.
See 'docker run --help'.
```

**Comment by xichenpan** [comment]
Hi @fikry102 , it seems you do not have NVIDIA Docker, can you try to install it?
maybe this gonna work
```
sudo apt-get update
sudo apt-get install nvidia-container-toolkit
sudo systemctl restart docker
```
